product-scenarios/scenarios-common/src/main/java/org/wso2/am/scenario/test/common/ScenarioTestBase.java

```diff
@@ -587,4 +624,89 @@ public boolean isTagsResponseContainsTag(JSONObject tagsResponse, String tagName
         }
         return false;
     }
+
+
+    /**
+     * This method can be used to wait for API deployment sync in distributed and clustered environment
+     * APIStatusMonitor will be invoked to get API related data and then verify that data matches with
+     * expected response provided.
+     *
+     * @param apiProvider      - Provider of the API
+     * @param apiName          - API name
+     * @param apiVersion       - API version
+     * @param expectedResponse - Expected response
+     * @throws APIManagerIntegrationTestException - Throws if something goes wrong
+     */
+    protected void waitForAPIDeploymentSync(String apiProvider, String apiName, String apiVersion,
+                                            String expectedResponse)
+            throws APIManagerIntegrationTestException {
+
+        long currentTime = System.currentTimeMillis();
+        long waitTime = currentTime + WAIT_TIME;
+        String colonSeparatedHeader = "admin" + ':' + "admin";
+        String authorizationHeader = "Basic "+new String(Base64.encodeBase64(colonSeparatedHeader.getBytes()));
+        Map headerMap = new HashMap();
+        headerMap.put("Authorization",authorizationHeader);
+        String tenantIdentifier = getTenantIdentifier(apiProvider);
+
+        while (waitTime > System.currentTimeMillis()) {
+            HttpResponse response = null;
+            try {
+                response = HttpClient.doGet(getBackendEndServiceEndPointHttps("") +
+                        "APIStatusMonitor/apiInformation/api/" +
+                        tenantIdentifier +
+                        apiName + "/" + apiVersion, headerMap);
+            } catch (IOException ignored) {
+                log.warn("WebAPP:" + " APIStatusMonitor not yet deployed or" + " API :" + apiName + " not yet " +
+                        "deployed " + " with provider: " + apiProvider);
+            }
+
+            log.info("WAIT for availability of API: " + apiName + " with version: " + apiVersion
+                    + " with provider: " + apiProvider + " with Tenant Identifier: " + tenantIdentifier
+                    + " with expected response : " + expectedResponse);
+
+            if (response != null) {
+                log.info("Data: " + response.getData());
+                if (response.getData().contains(expectedResponse)) {
+                    log.info("API :" + apiName + " with version: " + apiVersion +
+                            " with expected response " + expectedResponse + " found");
+                    break;
+                } else {
+                    try {
+                        Thread.sleep(500);
+                    } catch (InterruptedException ignored) {
+
```

Comment:
can we handle this exception in a proper way?

Reply:
@dinusha92 we need to ignore this exception. What we are doing here is trying to wait for API deployment in a loop. So we are pausing the loop for 500ms after checking the deployment status onece. Failure to sleep is ignored and no need to log/handle it. Anyway I'll add a comment in the code